server/s3_copy_object_action.cc

```diff
@@ -292,13 +303,76 @@ void S3CopyObjectAction::read_data_block_success() {
     return;
   }
   if (copy_failed) {
+    assert(!write_in_progress);
+
     send_response_to_s3_client();
     return;
   }
+  assert(data_blocks_read.empty());
+  data_blocks_read = motr_reader->extract_blocks_read();
+
+  size_t bytes_in_chunk_count = 0;
+
+  if (data_blocks_read.empty()) {
+    s3_log(S3_LOG_ERROR, stripped_request_id, "Motr reader returned no data");
+
+    s3_put_action_state = S3PutObjectActionState::writeFailed;
+    copy_failed = true;
+    set_s3_error("InternalError");
+
+    if (!write_in_progress) {
+      send_response_to_s3_client();
+    }
+    return;
+  }
+  // Calculating actial size of data that has just been read
+
+  for (size_t i = 0, n = data_blocks_read.size() - 1; i < n; ++i) {
+    const auto motr_block_size = data_blocks_read[i].second;
+    assert(motr_block_size == motr_unit_size);
+
+    // We can use multiplication, but something may change in the future
+    bytes_in_chunk_count += motr_block_size;
+  }
+  if (bytes_in_chunk_count >= bytes_left_to_read) {
+    s3_log(S3_LOG_ERROR, stripped_request_id, "Too many data has been read");
```

Comment:
debug message makes more send if we log value of bytes_in_chunk_count and bytes_left_to_read

Reply:
Agree. Will do